nautobot/dcim/models/devices.py

```diff
@@ -1619,11 +1619,16 @@ class Meta:
     def __str__(self):
         serial = f" (Serial: {self.serial})" if self.serial else ""
         asset_tag = f" (Asset Tag: {self.asset_tag})" if self.asset_tag else ""
-        display = str(self.module_type) + serial + asset_tag
+        return str(self.module_type) + serial + asset_tag
+
+    @property
+    def display(self):
         if self.location:
-            return f"{display} at location {self.location}"
+            return f"{self!s} at location {self.location}"
+        elif self.parent_module_bay.parent_device is not None:
+            return f"{self.module_type!s} installed in {self.parent_module_bay.parent_device.display}"
         else:
-            return f"{display} installed in {self.parent_module_bay}"
+            return f"{self.module_type!s} installed in {self.parent_module_bay.parent_module.display}"
```

Comment:
I wonder if these are overly verbose, or whether the clarity is worth the verbosity? some ideas for shorter alterniatives:

- `{self.location} : {self!s}`
- `{self.parent_module_bay.parent_device.display} : {self.module_type!s}`
- `{self.parent_module_bay.parent_module.display} : {self.module_type!s}`

Reply:
I find it nice to have it in the display field for more context in the UI:

![image](https://github.com/nautobot/nautobot/assets/75227981/220cf078-27a0-4dbb-8a2a-a6f6af5a2d12)

Instead of:

![image](https://github.com/nautobot/nautobot/assets/75227981/8c5a8110-b0ba-44bb-8681-34403ac707ce)